lang/java/reef-io/src/test/java/org/apache/reef/io/network/NetworkServiceTest.java

```diff
@@ -97,13 +98,24 @@ public void testMessagingNetworkService() throws Exception {
 
       LOG.log(Level.FINEST, "=== Test network service receiver start");
       LOG.log(Level.FINEST, "=== Test network service sender start");
-      try (final NameResolver nameResolver = injector2.getInstance(NameResolver.class);
-           final NetworkService<String> ns2 = new NetworkService<>(factory, 0, nameResolver,
-               new StringCodec(), injector.getInstance(MessagingTransportFactory.class),
-               new MessageHandler<String>(name2, monitor, numMessages), new ExceptionHandler(), localAddressProvider);
-           final NetworkService<String> ns1 = new NetworkService<>(factory, 0, nameResolver,
-               new StringCodec(), injector.getInstance(MessagingTransportFactory.class),
-               new MessageHandler<String>(name1, null, 0), new ExceptionHandler(), localAddressProvider)) {
+      try (final NameResolver nameResolver = injector2.getInstance(NameResolver.class)) {
+        injector2.bindVolatileParameter(NetworkServiceParameters.NetworkServiceIdentifierFactory.class, factory);
+        injector2.bindVolatileInstance(NameResolver.class, nameResolver);
+        injector2.bindVolatileParameter(NetworkServiceParameters.NetworkServiceCodec.class, new StringCodec());
+        injector2.bindVolatileParameter(NetworkServiceParameters.NetworkServiceTransportFactory.class,
+            injector.getInstance(MessagingTransportFactory.class));
+        injector2.bindVolatileParameter(NetworkServiceParameters.NetworkServiceExceptionHandler.class,
+            new ExceptionHandler());
+
+        final Injector injectorNs2 = injector2.forkInjector();
+        injectorNs2.bindVolatileParameter(NetworkServiceParameters.NetworkServiceHandler.class,
+            new MessageHandler<String>(name2, monitor, numMessages));
+        final NetworkService<String> ns2 = injectorNs2.getInstance(NetworkService.class);
+
+        final Injector injectorNs1 = injector2.forkInjector();
```

Comment:
Same thing here, create a new configuration and put it as a parameter in forkInjector().